packages/dumb/src/components/stats/loop-mode-stat/loop-mode-stat.tsx

```diff
@@ -0,0 +1,117 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { FunctionComponent, memo } from 'react'
+import styles from './loop-mode-stat.css'
+import { StatTooltip } from '../../tooltips/stat-tooltip/stat-tooltip'
+import { Box } from '@material-ui/core'
+import { LoopModePercentageDetail } from './loop-mode-percentage-detail'
+import { LoopModeLabel } from './loop-mode-label'
+
+interface LoopModeStatProps {
+  annotations: []
+  automated: number
+  hideTooltip: boolean
+  manual: number
+  title: string
+  total: number
+}
+
+const circleRadius = 70
+
+const LoopModeStat: FunctionComponent<LoopModeStatProps> = (props) => {
+  const { annotations, automated, hideTooltip, manual, title, total } = props
+
+  const automatedPercentage = Math.round(100 * automated / total)
+  const manualPercentage = Math.round(100 * manual / total)
+
+  const angle = manualPercentage * Math.PI / 100
+  const x = circleRadius * 2.0 * Math.cos(angle)
+  const y = -circleRadius * 2.0 * Math.sin(angle)
+
+  return (
+    <div data-testid="loop-mode-stat">
+      <Box className={styles.title}>
+        {title}
+        {!hideTooltip &&
+          <StatTooltip annotations={annotations} />
+        }
+      </Box>
+      <svg
+        xmlns="http://www.w3.org/2000/svg" viewBox="0 0 290 80"
+        className={`${styles.statWheelTimeInAuto} ${styles.donutChart}`}
+      >
+        <LoopModeLabel className={styles.onColor} transform="translate(10 15)" translationKey="wheel-label-on" />
+        <LoopModeLabel className={styles.offColor} transform="translate(240 15)" translationKey="wheel-label-off" />
+        <LoopModePercentageDetail
+          className={styles.labelOnValueUnits}
+          percentage={automatedPercentage}
+          transform="translate(30 63)"
+          value={automated}
+        />
+        <LoopModePercentageDetail
+          className={styles.labelOffValueUnits}
+          percentage={manualPercentage}
+          transform="translate(260 63)"
+          value={manual}
+        />
+        <g transform="translate(145 75)">
+          <mask id="half-wheel-mask">
+            <rect x="-70" y="-70" width="140" height="70" fill="white" />
+            <circle cx="0" cy="0" r="40" fill="black" />
+          </mask>
+          <clipPath id="half-circle-percent-clip">
+            {x && y
```

Comment:
I find this part very hard to read, in my opinion it would be more readable to compute it in a dedicated method. I know the convention is to return everything as flat as possible but still... 😬